### PR TITLE
ci: Introduce a weekly audit task

### LINKED ISSUE #257: Track Rustsec advisories
As a user of the sentry crates, I would appreciate it if you tracked Rustsec advisories for dependencies. This can be accomplished with a daily run of the [`audit-check`](https://github.com/actions-rs/audit-check) github action.

### PR DESCRIPTION
**Pull request by Swatinem**
Also cleans up the weekly clippy task as well

fixes #257

### CODE REVIEW

File: .github/workflows/weekly.yml [hunk 1]
```diff
@@ -1,14 +1,15 @@
-name: Weekly Beta
+name: Weekly CI
 
 on:
+  push: # TODO: remove before merging
```
Comment (flub) [inline .github/workflows/weekly.yml:4]: Please do :)

File: .github/workflows/weekly.yml [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
   schedule:
     - cron: "0 0 * * 1" # every monday at 00:00
 
 env:
   RUSTFLAGS: -Dwarnings
 
 jobs:
-  test:
+  weekly-clippy-beta:
```
Comment (flub) [inline .github/workflows/weekly.yml:12]: huh, is it this job name which must be unique across all workflows for the clippy action to get this right?

File: .github/workflows/weekly.yml [hunk 3]
```diff
@@ -24,9 +25,20 @@ jobs:
       - uses: actions-rs/clippy-check@v1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+          name: clippy-beta-weekly
```
Comment (flub) [inline .github/workflows/weekly.yml:28]: or this one?

**Review by flub — APPROVED** [review]

File: .github/workflows/weekly.yml (hunk 3 above)
Comment (Swatinem) [reply .github/workflows/weekly.yml:28]: this one here…

**Pull request merged**